Userland/Libraries/LibJS/Parser.h

```diff
@@ -297,5 +297,11 @@ class Parser {
     Vector<ParserState> m_saved_state;
     HashMap<Position, TokenMemoization, PositionKeyTraits> m_token_memoizations;
     Program::Type m_program_type;
+
+public:
+    // Hidden at the very end so noone will be tempted to use it. Seriously, please don't.
+    // This should only ever be needed by FunctionConstructor::create_dynamic_function_node() for some manual parser state shenanigans.
+    ParserState& state() { return m_state; };
```

Comment:
It is uglier but if it is really just one function which uses it we could just make this a friend?
Because I agree this state being visible can be dangerous.

Reply:
TIL you can add friend functions as well - done :)